### PR TITLE
Add function removeLayer for color, elevation and geometry layers

### LINKED ISSUE #906: Proposition : function removeLayer() for geometric layer
For now it is only possible to remove a 'color' layer. Could be nice to do the same with a geometric (geojson) layer.

### PR DESCRIPTION
**Pull request by gchoqueux**
## Description
New function `removeLayer` to remove color, elevation and geometry layers from viewer

Add tests on `removeLayer`.

fix #906

### CODE REVIEW

File: src/Core/View.js [hunk 1]
```diff
@@ -292,6 +295,82 @@ View.prototype.addLayer = function addLayer(layer, parentLayer) {
...
+
+/**
+ * Removes a specific imagery layer from the current layer list. This removes layers inserted with attach().
+ * @example
+ * view.removeLayer('layerId');
+ * @param      {string}   layerId      The identifier
+ * @return     {boolean}
+ */
+View.prototype.removeLayer = function removeLayer(layerId) {
+    const layer = this.getLayers(l => l.id === layerId)[0];
```
Comment (zarov) [inline src/Core/View.js]: General comment: we need a method to access directly a layer from a view, I can't look anymore at this line

File: src/Core/View.js [hunk 2]
```diff
@@ -292,6 +295,82 @@ View.prototype.addLayer = function addLayer(layer, parentLayer) {
     });
 };
 
+function disposeMesh(obj) {
```
Comment (zarov) [inline src/Core/View.js]: This should be in `GeometryLayer`, WDYT ?

File: src/Core/View.js (hunk 1 above)
Comment (gchoqueux) [reply src/Core/View.js]: yes, completely agree

File: src/Core/View.js (hunk 2 above)
Comment (gchoqueux) [reply src/Core/View.js]: yes, I will move it.

Comment (gchoqueux) [reply src/Core/View.js]: In thinking about it, I think that we should do the removing/disposing nodes in the `detach` method of the layer. I'll soon make a new proposal.

File: src/Core/View.js (hunk 1 above)
Comment (gchoqueux) [reply src/Core/View.js]: function `View.getLayerById` added

File: src/Core/View.js (hunk 2 above)
Comment (gchoqueux) [reply src/Core/View.js]: A new proposal is pushed.

File: src/Core/View.js [hunk 3]
```diff
@@ -340,6 +386,21 @@ View.prototype.getLayers = function getLayers(filter) {
...
+/**
+ * Gets the layer by identifier.
+ *
+ * @param {String}  layerId  The layer identifier
+ * @return {Layer}  The layer by identifier.
+ */
+
+View.prototype.getLayerById = function getLayers(layerId) {
+    const layers = this.getLayers(l => l.id === layerId);
+    if (layers.length === 1) {
```
Comment (zarov) [inline src/Core/View.js]: Unnecessary ? It should be impossible to have two layers with the same id.

File: src/Layer/ColorLayer.js [hunk 4]
```diff
@@ -73,6 +73,15 @@ class ColorLayer extends Layer {
     convert(data, extentDestination) {
         return textureConverter.convert(data, extentDestination, this);
     }
+
+    /**
+    * All layer's textures are removed from scene and disposed from video device.
+    */
+    removeDisposeNodes() {
```
Comment (zarov) [inline src/Layer/ColorLayer.js]: Maybe add a "fake" method in `Layer` as well: either doing nothing, or printing a warning that this method needs to be implemented (like in `Source`).

File: src/Core/View.js [hunk 5]
```diff
@@ -340,6 +386,21 @@ View.prototype.getLayers = function getLayers(filter) {
...
 };
 
+/**
+ * Gets the layer by identifier.
+ *
+ * @param {String}  layerId  The layer identifier
+ * @return {Layer}  The layer by identifier.
+ */
+
+View.prototype.getLayerById = function getLayers(layerId) {
```
Comment (zarov) [inline src/Core/View.js]: `function getLayerById` instead.

Also we need to change (in next PR) everywhere this method could be used.

File: src/Layer/ColorLayer.js (hunk 4 above)
Comment (zarov) [reply src/Layer/ColorLayer.js]: Also this seems like a weird name to understand

File: src/Core/View.js (hunk 5 above)
Comment (gchoqueux) [reply src/Core/View.js]: done

File: src/Core/View.js (hunk 3 above)
Comment (gchoqueux) [reply src/Core/View.js]: done

File: src/Layer/ColorLayer.js (hunk 4 above)
Comment (gchoqueux) [reply src/Layer/ColorLayer.js]: done

Comment (gchoqueux) [reply src/Layer/ColorLayer.js]: I rename to `Layer.delete`

**Comment by gchoqueux** [comment]
@zarov  you could review update.
 I need to implement delete function for `OrientedImageLayer`.

**Review by zarov — APPROVED** [review]
LGTM

**Pull request merged**